adapters/ds/gst_plugins/python/always_on_rtsp_frame_processor.py

```diff
@@ -182,26 +280,57 @@ def crop_to_fit(
             target_width,
             target_height,
         )
-        output_frame.setTo((0, 0, 0, 0))
-        source = cv2.cuda.GpuMat(
-            input_frame,
-            (
-                (in_width - target_width) // 2,  # left
-                (in_height - target_height) // 2,  # top
-                target_width,  # width
-                target_height,  # height
-            ),
-        )
-        target = cv2.cuda.GpuMat(
-            output_frame,
-            (
-                (out_width - target_width) // 2,  # left
-                (out_height - target_height) // 2,  # top
-                target_width,  # width
-                target_height,  # height
-            ),
-        )
-        source.copyTo(target)
+
+        in_left = (in_width - target_width) // 2
+        in_top = (in_height - target_height) // 2
+        out_left = (out_width - target_width) // 2
+        out_top = (out_height - target_height) // 2
+        if isinstance(input_frame, cv2.cuda.GpuMat):
+            output_frame.setTo((0, 0, 0, 0))
+            source = cv2.cuda.GpuMat(
+                input_frame,
+                (in_left, in_top, target_width, target_height),
+            )
+            target = cv2.cuda.GpuMat(
+                output_frame,
+                (out_left, out_top, target_width, target_height),
+            )
+            source.copyTo(target)
+
+        else:
+            output_frame.fill(0)
+            output_frame[
+                out_top : out_top + target_height,
+                out_left : out_left + target_width,
+            ] = input_frame[
+                in_top : in_top + target_height,
+                in_left : in_left + target_width,
+            ]
+
+        return Gst.FlowReturn.OK
+
+    @contextmanager
+    def _gst_buffer_to_np_array(self, buffer: Gst.Buffer):
+        """Convert Gst.Buffer to numpy array.
+        All the changes will be applied to the buffer.
+        """
+
+        is_mapped, map_info = buffer.map(Gst.MapFlags.READ | Gst.MapFlags.WRITE)
```

Comment:
There are ctypes ffi and `map_gst_buffer` in `savant/gstreamer`. Are they still relevant? - Then please use (modify and use) them, otherwise delete them.

Reply:
Used `map_gst_buffer`. Now we don't need to copy memory to make a writable numpy buffer.